### PR TITLE
GH-16: Add ability to unlock user with an email address.

### PR DESCRIPTION
**Pull request by the-t-in-rtf**
@jobara pointed out a key assumption in the utils that prevents logging in using only a username.  This pull should address that.

### CODE REVIEW

File: tests/js/server/forgot-tests.js [hunk 1]
```diff
@@ -157,6 +180,33 @@ fluid.defaults("fluid.tests.express.user.reset.caseHolder", {
...
+                        //    event: "{fullResetForgotRequest}.events.onComplete",
+                        //    args: ["{fullResetForgotRequest}", "{fullResetForgotRequest}.nativeResponse", "{arguments}.0", 200]
+                        //},
+                        {
+                            listener: "fluid.tests.express.user.reset.caseHolder.fullResetExtractCodeFromEmailAndReset",
+                            event:    "{testEnvironment}.smtp.mailServer.events.onMessageReceived",
+                            args:     ["{testEnvironment}", "{emailResetResetRequest}"] // testEnvironment, resetRequest
+                        },
+                        {
+                            listener: "fluid.tests.express.user.reset.caseHolder.verifyResponse",
```
Comment (amb26) [inline tests/js/server/forgot-tests.js:204]: I assume there is knowledge that the "onComplete" event must definitely follow the "onMessageReceived" event but a comment about the location of this knowledge would be helpful

**Comment by amb26** [comment]
Somehow I've lost the ability to run these tests successfully - they jam at the webdriver initialisation point:
````
13:47:47.126:  Express started...

DevTools listening on ws://127.0.0.1:49550/devtools/browser/733988df-4780-4f34-8164-43a9cc0b53d5
13:47:55.680:  Test case listener has not responded after 5000ms - at sequence pos 5 of 15 sequence element {
    "event": "{testEnvironment}.webdriver.events.onWaitComplete",
    "listener": "{testEnvironment}.webdriver.actionsHelper",
    "args": [
        [
            {
                "fn": "sendKeys",
                "args": [
                    "?",
                    "existing@localhost",
                    "?",
                    "?"
                ]
            }
        ]
    ]
} of fixture chrome: Confirm that passwords must match...
````
I can confirm that I also can't run tests in main which I could last night, so there is probably nothing wrong with this branch - all the same I'd appreciate someone else running the tests independently - perhaps @jobara ? Cheers

**Comment by jobara** [comment]
I was able to get all of the tests to run. I didn't have ChromeDriver installed, so I had to go through the process of doing that. @amb26 I wonder if you may need to update the version of ChromeDriver installed on your machine. I'm wondering if there might be some incompatibilities between the version of ChromeDriver and Chrome that you have installed.

**Comment by amb26** [comment]
Yes, I regularly update my ChromeDriver since if there is a version mismatch it fails pretty early. It claims to have a version matching my Chrome so perhaps I have another issue on my machine. Thanks for checking out the tests -

**Comment by the-t-in-rtf** [comment]
Thanks, @amb26.  I have cut the full release of 2.0.1 and added a release page on GitHub.

**Pull request merged**